CONTRIBUTING.md

```diff
@@ -312,6 +312,41 @@ Alternatively, the `make test_ci` target can be used instead of
 `make test` in order to run in a setup that replicates the configuration
 we used in our CI systems more closely.
 
+### Snapshot Testing for Visualizations
+
+If you are working on code that makes changes to any matplotlib visualisations 
+you will need to check that your changes don't break any snapshot tests, and add 
+new tests where necessary. You can do this as follows:
+
+1. Make sure you have pushed your branch with the latest changes.
+2. Go to link: `https://mybinder.org/v2/gh/<github_user>/qiskit-terra/<branch>?urlpath=apps/test/ipynb/mpl_tester.ipynb`. 
+This opens a jupyter application running in the cloud that automatically runs 
+the snapshot tests (note this may take some time to finish loading).
+3. If any tests fail you should see sets of 3 images for each failed test 
+(left: reference image, middle: your test result, right: differences):
+   <img width="995" alt="Screenshot_2021-03-26_at_14 13 54" src="https://user-images.githubusercontent.com/23662430/112663508-d363e800-8e50-11eb-9478-6d665d0ff086.png">
+4. Fix any broken tests. Working on code for one aspect of the visualisations 
+can sometimes result in minor changes elsewhere to spacing etc. In these cases 
+you just need to update the reference images as follows:
+    - download the mismatched images (link at top of jupyter notebook output)
+    - unzip the folder
+    - copy and paste the new images into `qiskit-terra/test/ipynb/mpl/references`, 
+  replacing the existing reference images
+    - commit and push your changes, restart the jupyter app in your browser. The 
```

Comment:
```suggestion
    - add, commit, and push your changes, restart the Jupyter Notebook app in your browser. The 
```

Reply:
Not fully sure if the binder env will update the branch. You might need to create a new env by visiting the link in step 2. This needs to be confirmed.